app/models/synonym.rb

```diff
@@ -30,38 +30,70 @@
 ################################################################################
 
 class Synonym < AbstractModel
+  require "arel-helpers"
+
+  include ArelHelpers::ArelTable
+
   has_many :names
 
-  # Mightly cronjob to ensure that no synonym records accidentally got deleted.
+  # Nightly cronjob to ensure that no synonym records accidentally got deleted.
   # This actually happened to Fungi itself(!)  Not sure how it happened, and
   # obviously I'd prefer to fix the cause.  But meanwhile, might as well keep
   # the site working...
   def self.make_sure_all_referenced_synonyms_exist
     msgs = []
-    references = Name.connection.select_values(%(
-      SELECT DISTINCT synonym_id FROM names
-      WHERE synonym_id IS NOT NULL
-      ORDER BY synonym_id ASC
-    ))
-    records = Name.connection.select_values(%(
-      SELECT id FROM synonyms ORDER BY id ASC
-    ))
+    # reference_select = arel_select_referenced_synonyms
+    # # puts(reference_select.to_sql)
+    # references = Name.connection.select_values(reference_select.to_sql)
+    references = Name.select(:synonym_id).where.not(synonym: nil).distinct.
+                 order(:synonym_id).pluck(:synonym_id)
+    # puts(references.join(",").to_s)
+
+    # SELECT id FROM synonyms ORDER BY id ASC
+    # record_select = Synonym.select(:id).order(Synonym[:id].asc)
+    # puts(record_select.to_sql)
+    # records = Name.connection.select_values(record_select.to_sql)
+    records = Synonym.all.order(id: :asc).pluck(:id)
+    # puts(records.join(",").to_s)
     unused  = records - references
     missing = references - records
+    # puts(unused.join(",").to_s)
+    # puts(missing.join(",").to_s)
+
     if unused.any?
-      Name.connection.execute(%(
-        DELETE FROM synonyms
-        WHERE id IN (#{unused.map(&:to_s).join(",")})
-      ))
+      Synonym.where(id: unused).delete_all
+
       msgs << "Deleting #{unused.count} unused synonyms: #{unused.inspect}"
     end
     if missing.any?
-      Name.connection.execute(%(
-        INSERT INTO synonyms (id) VALUES
-        #{missing.map { |id| "(#{id})" }.join(",")}
-      ))
+      insert_manager = arel_insert_missing_synonyms(missing)
+      # puts(insert_manager.to_sql)
+      Name.connection.execute(insert_manager.to_sql)
```

Comment:
```suggestion
      Name.connection.execute(arel_insert_all(missing).to_sql)
```
Reads more like both (a) @pellaea's original SQL and (b) Rails 6 `insert_all`.
Therefore easier to understand the change, understand the new code, eventually switch to `insert_all`.